### DISCUSSION TITLE
Add the ability to select all options

### DISCUSSION THREAD

**Discussion by CodeGradox** [brianvoe/slim-select#514] [Ideas]
Hi! We're currently using [tom-select](https://tom-select.js.org/) with some custom plugins, but we have encountered some performance issues when it comes to selecting many elements at the same time

I see that slim-select allows the user to select all elements in an optgroup, which is great! However, we want the ability so select and deselect all elements in the select. We were able to create a plugin for this in tom-select, but it freezes the browsers if it tries to select over 1000 elements.

We basically need something like this [feature found in bootstrap-select](https://developer.snapappointments.com/bootstrap-select/examples/#selectdeselect-all-options).

**Comment by brianvoe** [comment]
selectAll is currently only on groups right now. If you want to submit a pr for all ill take a look at it.

**Comment by CodeGradox** [comment]
We found another library with support for selectAll today: [virtual-select](https://sa-si-dev.github.io/virtual-select/) today 🙈

Sorry for the inconvenience 😅
